src/richie/apps/courses/factories.py

```diff
@@ -141,21 +187,124 @@ def extended_object(self):
+            (
+                "Open edX Conference 2018 Opening and Welcome remarks",
+                "//www.youtube.com/embed/zzx6MgBAbCc",
+            ),
+        )
+
+        label, url = random.choice(available_videos)
+
+        return add_plugin(
+            language=language,
+            placeholder=placeholder,
+            plugin_type="VideoPlayerPlugin",
+            label=label,
+            embed_link=url,
+        )
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_syllabus(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for syllabus with a long random text
+        """
+        create_text_plugin(self, "course_syllabus", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_format(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for course format with a long random text
+        """
+        create_text_plugin(self, "course_format", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_prerequisites(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for pre requisites with a long random text
+        """
+        create_text_plugin(self, "course_prerequisites", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_plan(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for plan with a long random text
+        """
+        create_text_plugin(self, "course_plan", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_content_license(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for course content license with a long random text
+        """
+        create_text_plugin(self, "license_course_content", nb_paragraphs=1)
+
+    @factory.post_generation
+    # pylint: disable=unused-argument
+    def fill_participation_license(self, create, extracted, **kwargs):
+        """
+        Add a text plugin for course participation license with a long random text
+        """
+        create_text_plugin(self, "license_course_participation", nb_paragraphs=1)
 
     @factory.post_generation
     # pylint: disable=unused-argument, attribute-defined-outside-init, no-member
     def with_subjects(self, create, extracted, **kwargs):
-        """Add subjects to ManyToMany relation."""
+        """
+        Add subjects to ManyToMany relation.
+        """
         if create and extracted:
-            self.subjects.set(extracted)
+            # First try to reach possible relations passed through kwargs when
+            # using this method from a ``Factory.create_batch(...)`` then
+            # switch to the basic method usage from a single Factory.
+            instances = kwargs.get("items", [])
+            weight = kwargs.get("weight", 0)
```

Comment:
Weight is the number of relation you want to allow, if you have 10 available relation but just want to give 3 to your object, you specify `weight=3`, it will randomly 3 of the 10 available objects

Reply:
IC. Should be called "size" ou "sample_size" ?